### PR TITLE
Integrate `scanResultBlob` pattern on robonode side

### PR DESCRIPTION
**Pull request by dmitrylavrenov**
Agreed on decoupling logic of #1026

### CODE REVIEW

**Comment by dmitrylavrenov** [comment]
Agreed on decoupling logic of #1026

**Review by MOZGIII — APPROVED** [review]

**Review by MOZGIII — CHANGES_REQUESTED** [review]
Actually hold on

**Review by MOZGIII — CHANGES_REQUESTED** [review]
Add tests to ensure errors return `scan_result_blob` too

File: crates/robonode-server/src/logic/mod.rs [hunk 1]
```diff
@@ -20,6 +20,9 @@ pub mod traits;
 
 pub use traits::*;
 
+/// A type alias representing a scan result blob.
+pub(crate) type ScanResultBlob = Option<String>;
```
Comment (MOZGIII) [inline crates/robonode-server/src/logic/mod.rs:24]:
```suggestion
pub(crate) type ScanResultBlob = String;
```

Applying option like this is very confusing when it comes to reading the code, better apply it at the usage location: `Option<ScanResultBlob>`

Comment (dmitrylavrenov) [reply crates/robonode-server/src/logic/mod.rs]: https://github.com/humanode-network/humanode/pull/1037/commits/33f49d52f6000f24f1c3ffb850fa4a23a8e86cd7

**Comment by dmitrylavrenov** [comment]
Closes in favour of #1065

**Pull request closed without merging**